### PR TITLE
NUI-1582: Export shared SDK code from asset-compute-pipeline

### PR DESCRIPTION
**Pull request by sthyagarajan**
Export shared SDK code from asset-compute-pipeline
Fixes https://jira.corp.adobe.com/browse/NUI-1582

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

Fixes # (issue)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/storage/index.js [hunk 1]
```diff
@@ -26,146 +26,144 @@ const validDataUrl = require('valid-data-url');
...
     }
 }
 
-module.exports = {
-    getSource,
-    putRendition,
-    getAsset
-};
\ No newline at end of file
+module.exports = Storage;
```
Comment (jdelbick) [inline lib/storage/index.js:169]: nitpick, do we want to export it with `{}` so we can import it internally the same way we import it from external files:
internal
```js
const Storage = require('../../lib/storage'); // current
const {Storage} = require('../../lib/storage');
```

external
```js
const { Storage } = require('@adobe/asset-compute-pipeline');
```

Comment (jdelbick) [reply lib/storage/index.js:169]: I don't have strong opinions about this, just wanted to bring it up in case others did

Comment (sthyagarajan) [reply lib/storage/index.js:169]: Changed it to keep it consistent

File: index.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ const Action = require("./lib/action");
 const Asset = require("./lib/asset");
 const Prepare = require("./lib/prepare");
 const Rendition = require("./lib/rendition");
-const Storage = require("./lib/storage");
+const { Storage } = require("./lib/storage");
```
Comment (tmathern) [inline index.js:19]: If we do this here, we should do it for all the other imports too... (Prepare, Rendition, etc)

File: lib/storage/index.js [hunk 3]
```diff
@@ -26,146 +26,144 @@ const validDataUrl = require('valid-data-url');
...
     }
-
-    if (source.url && validUrl.isUri(source.url)) {
-        const basename = path.basename(new URL(source.url).pathname);
-        return  `${SOURCE_BASENAME}${extension(basename, source.mimeType)}`;
+    
+    // There is at least one worker (graphics magick) that in some cases depends
+    // upon the file extension so it is best to try to use the appropriate one
+    // based on the filename, url, or mimetype
+    static sourceFilename(source) {
```
Comment (tmathern) [inline lib/storage/index.js:115]: Either extension is a property or a function.
If a property, it should be static get sourceFilename(_, _).
If a function, it should be called getSourceFilename.

I would go for the property.

**Pull request merged**